Makefile.rules

```diff
@@ -0,0 +1,162 @@
+CXXFLAGS += -O2 -fPIC -Wall
```

Comment:
Several flags got lost along the way: `--std=c++17 -Wpedantic - Wextra -Wno-unused-parameter -Werror -Wfatal-errors -gdwarf-4 -DJLM_DEBUG -DJLM_ENABLE_ASSERTS`